packages/fast-components/src/slider/slider.ts

```diff
@@ -0,0 +1,320 @@
+
+    /**
+     * The element's value to be included in form submission changed.
+     */
+    @attr
+    public value: string; // Map to proxy element.
+    private valueChanged(): void {
+        if (Number(this.value) === Number.NaN) {
+            this.value = "1";
+        }
+
+        if (this.proxy instanceof HTMLElement) {
+            this.updateForm();
+        }
+
+        const percentage: number =
+            this.direction !== Direction.rtl
+                ? (1 - Number(this.value) / (Number(this.max) - Number(this.min))) * 100
+                : (Number(this.value) / (Number(this.max) - Number(this.min))) * 100;
+
+        if (this.orientation === SliderOrientation.horizontal) {
+            this.position = this.isDragging
+                ? `right: ${percentage}%; transition: all 0.1s ease;`
+                : `right: ${percentage}%; transition: all 0.2s ease;`;
+        } else {
+            this.position = this.isDragging
+                ? `bottom: ${percentage}%; transition: all 0.1s ease;`
+                : `bottom: ${percentage}%; transition: all 0.2s ease;`;
+        }
+
+        this.$emit("change");
+    }
+
+    /**
+     * Min allowed value default is 0
+     */
+    @attr({ converter: nullableNumberConverter })
+    public min: number = 0; // Map to proxy element.
+    private minChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.min = `${this.min}`;
+        }
+    }
+
+    /**
+     * Max allowed value default is 10
+     */
+    @attr({ converter: nullableNumberConverter })
+    public max: number = 10; // Map to proxy element.
+    private maxChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.max = `${this.max}`;
+        }
+    }
+
+    /**
+     * Value to increment or decrement via arrow keys, mouse click or drag
+     */
+    @attr({ converter: nullableNumberConverter })
+    public step: number = 1; // Map to proxy element.
+    private stepChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.step = `${this.step}`;
+        }
+    }
+
+    /**
+     * Orientation value, horizontal | vertical
+     */
+    @attr
+    public orientation: SliderOrientation = SliderOrientation.horizontal;
+    private orientationChanged(): void {
+        if (this.orientation === SliderOrientation.horizontal) {
+            this.classList.remove("slider-vertical");
+            this.classList.add("slider-horizontal");
+        } else {
+            this.classList.remove("slider-horizontal");
+            this.classList.add("slider-vertical");
+        }
+    }
+
+    /**
+     * mode value, default singleValue
+     */
+    @attr
+    public mode: SliderMode = SliderMode.singleValue;
+
+    protected proxy = document.createElement("input");
+
+    constructor() {
+        super();
+        this.proxy.setAttribute("type", "range");
+        this.direction = this.getDirection();
```

Comment:
The `getDirection()` function uses parentElement, which might be null if someone creates this in JS and then later add it to the DOM. This is something you'll want to do in the connected callback.

Reply:
moved to connectedCallback and if rtl trigger setting value again to get the right calculation result